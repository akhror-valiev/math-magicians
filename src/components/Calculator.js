import './calculator.css';
import '../logic/operate';
import PropTypes from 'prop-types';

const Calculator = (props) => {
  const { calcStates, handleClick } = props;
  const { total, next } = calcStates;
  let val = 0;
  if (next) {
    val = next;
  } else if (total) {
    val = total;
  }

  return (
    <div className="main-container">
      <input
        className="grid-container display"
        placeholder="0"
        value={val}
        readOnly
      />
      <button type="button" className="grid-container AC" onClick={handleClick}>
        AC
      </button>
      <button
        type="button"
        className="grid-container operations"
        onClick={handleClick}
      >
        +/-
      </button>
      <button
        type="button"
        className="grid-container procentage"
        onClick={handleClick}
      >
        %
      </button>
      <button
        type="button"
        className="grid-container divide operation"
        onClick={handleClick}
      >
        รท
      </button>
      <button type="button" className="grid-container seven" onClick={handleClick}>
        7
      </button>
      <button type="button" className="grid-container eight" onClick={handleClick}>
        8
      </button>
      <button type="button" className="grid-container nine" onClick={handleClick}>
        9
      </button>
      <button
        type="button"
        className="grid-container multiply operation"
        onClick={handleClick}
      >
        x
      </button>
      <button type="button" className="grid-container four" onClick={handleClick}>
        4
      </button>
      <button type="button" className="grid-container five" onClick={handleClick}>
        5
      </button>
      <button type="button" className="grid-container six" onClick={handleClick}>
        6
      </button>
      <button
        type="button"
        className="grid-container minus operation"
        onClick={handleClick}
      >
        -
      </button>
      <button type="button" className="grid-container one" onClick={handleClick}>
        1
      </button>
      <button type="button" className="grid-container two" onClick={handleClick}>
        2
      </button>
      <button type="button" className="grid-container three" onClick={handleClick}>
        3
      </button>
      <button
        type="button"
        className="grid-container plus operation"
        onClick={handleClick}
      >
        +
      </button>
      <button type="button" className="grid-container zero" onClick={handleClick}>
        0
      </button>
      <button type="button" className="grid-container period" onClick={handleClick}>
        .
      </button>
      <button
        type="button"
        className="grid-container equal operation"
        onClick={handleClick}
      >
        =
      </button>
    </div>
  );
};

Calculator.propTypes = {
  calcStates: PropTypes.instanceOf(Object).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Calculator;
