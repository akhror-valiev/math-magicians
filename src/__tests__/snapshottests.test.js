import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';
import Home from '../components/Home';

describe('Snapshot for Calculator and Home', () => {
  it('Snapshot should match component render', () => {
    const mycalculator = render(<Calculator />);
    expect(mycalculator).toMatchSnapshot();
  });

  it('Snapshot should match component render', () => {
    const homeview = render(<Home />);
    expect(homeview).toMatchSnapshot();
  });
});
