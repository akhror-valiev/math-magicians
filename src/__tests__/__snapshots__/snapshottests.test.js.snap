// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for Calculator and Home Snapshot should match component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-container"
      >
        <input
          class="grid-container display"
          placeholder="0"
          readonly=""
          value="0"
        />
        <button
          class="grid-container AC"
          type="button"
        >
          AC
        </button>
        <button
          class="grid-container operations"
          type="button"
        >
          +/-
        </button>
        <button
          class="grid-container procentage"
          type="button"
        >
          %
        </button>
        <button
          class="grid-container divide operation"
          type="button"
        >
          ÷
        </button>
        <button
          class="grid-container seven"
          type="button"
        >
          7
        </button>
        <button
          class="grid-container eight"
          type="button"
        >
          8
        </button>
        <button
          class="grid-container nine"
          type="button"
        >
          9
        </button>
        <button
          class="grid-container multiply operation"
          type="button"
        >
          ×
        </button>
        <button
          class="grid-container four"
          type="button"
        >
          4
        </button>
        <button
          class="grid-container five"
          type="button"
        >
          5
        </button>
        <button
          class="grid-container six"
          type="button"
        >
          6
        </button>
        <button
          class="grid-container minus operation"
          type="button"
        >
          -
        </button>
        <button
          class="grid-container one"
          type="button"
        >
          1
        </button>
        <button
          class="grid-container two"
          type="button"
        >
          2
        </button>
        <button
          class="grid-container three"
          type="button"
        >
          3
        </button>
        <button
          class="grid-container plus operation"
          type="button"
        >
          +
        </button>
        <button
          class="grid-container zero"
          type="button"
        >
          0
        </button>
        <button
          class="grid-container period"
          type="button"
        >
          .
        </button>
        <button
          class="grid-container equal operation"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-container"
    >
      <input
        class="grid-container display"
        placeholder="0"
        readonly=""
        value="0"
      />
      <button
        class="grid-container AC"
        type="button"
      >
        AC
      </button>
      <button
        class="grid-container operations"
        type="button"
      >
        +/-
      </button>
      <button
        class="grid-container procentage"
        type="button"
      >
        %
      </button>
      <button
        class="grid-container divide operation"
        type="button"
      >
        ÷
      </button>
      <button
        class="grid-container seven"
        type="button"
      >
        7
      </button>
      <button
        class="grid-container eight"
        type="button"
      >
        8
      </button>
      <button
        class="grid-container nine"
        type="button"
      >
        9
      </button>
      <button
        class="grid-container multiply operation"
        type="button"
      >
        ×
      </button>
      <button
        class="grid-container four"
        type="button"
      >
        4
      </button>
      <button
        class="grid-container five"
        type="button"
      >
        5
      </button>
      <button
        class="grid-container six"
        type="button"
      >
        6
      </button>
      <button
        class="grid-container minus operation"
        type="button"
      >
        -
      </button>
      <button
        class="grid-container one"
        type="button"
      >
        1
      </button>
      <button
        class="grid-container two"
        type="button"
      >
        2
      </button>
      <button
        class="grid-container three"
        type="button"
      >
        3
      </button>
      <button
        class="grid-container plus operation"
        type="button"
      >
        +
      </button>
      <button
        class="grid-container zero"
        type="button"
      >
        0
      </button>
      <button
        class="grid-container period"
        type="button"
      >
        .
      </button>
      <button
        class="grid-container equal operation"
        type="button"
      >
        =
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot for Calculator and Home Snapshot should match component render 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1>
          Home Page
        </h1>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Adipisci accusantium, similique at, aliquid nemo non consequatur corrupti quas ab doloribus nesciunt minus corporis? Obcaecati, dicta ea fugiat iste saepe iure!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1>
        Home Page
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Adipisci accusantium, similique at, aliquid nemo non consequatur corrupti quas ab doloribus nesciunt minus corporis? Obcaecati, dicta ea fugiat iste saepe iure!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
